src/amp.jl

```diff
@@ -226,80 +234,134 @@ function add_adaptive_partition(m::PODNonlinearModel; kwargs...)
+            for j in 1:λCnt
+                if point >= discretization[i][j] && point <= discretization[i][j+1]  # Locating the right location
+                    radius = calculate_radius(discretization[i], j, ratio)
+                    insert_partition(m, i, j, point, radius, discretization[i])
+                    push!(processed, i)
+                    break
                 end
-                if lb_new > lb_local && !isapprox(lb_new, lb_local; atol=m.disc_abs_width_tol) && abs(lb_new-lb_local)/(1e-8+abs(lb_local)) > m.disc_rel_width_tol # Insert new LB-based partition
-                    insert!(discretization[i], j+1, lb_new)
-                    lb_touch = false
-                end
-                if (ub_touch && lb_touch) || (m.disc_consecutive_forbid>0 && check_solution_history(m, i))
-                    distance = -1.0
-                    pos = -1
-                    for j in 2:length(discretization[i])  # it is made sure there should be at least two partitions
-                        if (discretization[i][j] - discretization[i][j-1]) > distance
-                            lb_local = discretization[i][j-1]
-                            ub_local = discretization[i][j]
-                            distance = ub_local - lb_local
-                            point = lb_local + (ub_local - lb_local) / 2   # reset point
-                            pos = j
-                        end
-                    end
-                    chunk = (ub_local - lb_local)/2
-                    insert!(discretization[i], pos, lb_local + chunk)
-                    (m.loglevel > 99) && println("[DEBUG] !DIVERT! VAR$(i): |$(lb_local) | 2 SEGMENTS | $(ub_local)|")
-                else
-                    (m.loglevel > 99) && println("[DEBUG] VAR$(i): SOL=$(round(point,4)) RATIO=$(ratio), PARTITIONS=$(length(discretization[i])-1)  |$(round(lb_local,4)) |$(round(lb_new,6)) <- * -> $(round(ub_new,6))| $(round(ub_local,4))|")
+            end
+        elseif m.var_type[i] == :Bin # This should never happen
+            warn("Binary variable in m.disc_vars. Check out what is wrong...")
+            continue  # No partition should be added to binary variable unless user specified
+        elseif m.var_type[i] == :Int
+            i in processed && continue
+            point = discover_int_point(m, i, discretization[i], point)
+            point == nothing && continue
+            for j in 1:λCnt
+                if point >= discretization[i][j] && point <= discretization[i][j+1]
+                    insert_partition(m, i, j, point, 0.5, discretization[i])
+                    push!(processed, i)
+                    break
                 end
-                break
             end
+        else
+            error("Unexpected variable types during injecting partitions")
         end
     end
 
     return discretization
 end
 
+function correct_point(m::PODNonlinearModel, partvec::Vector, point::Float64, var::Int)
```

Comment:
what does this function do  - can you add comments

Reply:
This function targets to address unexpected numerical issues by adding partitions to tight partitions. I added a line of comment for this function.